WalletWasabi/TorSocks5/TorProcessManager.cs

```diff
@@ -72,6 +73,21 @@ public async Task StartAsync()
 							ZipFile.ExtractToDirectory(Path.Combine(torDaemonsDir, "tor-win32.zip"), torDir);
 						}
 					}
+					else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+					{
+						try
+						{
+							ZipFile.ExtractToDirectory(Path.Combine(torDaemonsDir, "tor-linux64.zip"), torDir);
+						}
+						catch (UnauthorizedAccessException)
+						{
+							await Task.Delay(100);
+							ZipFile.ExtractToDirectory(Path.Combine(torDaemonsDir, "tor-linux64.zip"), torDir);
+						}
+
+						// Make sure there's sufficient permission.
+						EnvironmentHelpers.BashExec($"chmod -R 777 {torDir}");
```

Comment:
I don't know exacty what permissions this feature needs but I am sure `777` is excessive. Why should everybody be able to write in that folder?

Reply:
IDK, because Linux sucks? Honestly I have no idea what kind of permissions should I give here and not even sure how to figure that out.